src/components/FilterDropdown/index.jsx

```diff
@@ -30,17 +30,17 @@ const FilterDropdown = ({
     }),
     tooltipProps: isGroup && {
       content: "Add filter to this group",
-      placement: "bottom",
+      position: "bottom",
     },
   };
 
-  const onClose = () => {
-    setSelectedNode(node);
-    setIsFilterDropdownOpen(false);
-  };
-
   const handleClick = () => {
-    isFilterDropdownOpen ? onClose() : setIsFilterDropdownOpen(true);
+    if (isFilterDropdownOpen) {
+      setSelectedNode(node);
+      setIsFilterDropdownOpen(false);
+    } else {
+      setIsFilterDropdownOpen(true);
+    }
```

Comment:
This is safer from race conditions
```suggestion
    setIsFilterDropdownOpen(isOpen => {
      if (isOpen) setSelectedNode(node);
      return !isOpen;
    });
```